deployment/environment/memory/node.go

```diff
@@ -68,6 +70,115 @@ func (n Node) ReplayLogs(chains map[uint64]uint64) error {
+func (n Node) DeploymentNode() (deployment.Node, error) {
+	jdChainConfigs, err := n.JDChainConfigs()
+	if err != nil {
+		return deployment.Node{}, err
+	}
+	selMap, err := deployment.ChainConfigsToOCRConfig(jdChainConfigs)
+	if err != nil {
+		return deployment.Node{}, err
+	}
+	// arbitrarily set the first evm chain as the transmitter
+	var admin string
+	for _, k := range n.Keys.Transmitters {
+		admin = k
+		break
+	}
+	return deployment.Node{
+		NodeID:         n.Keys.PeerID.String(),
+		Name:           n.Keys.PeerID.String(),
+		SelToOCRConfig: selMap,
+		CSAKey:         n.Keys.CSA.ID(),
+		PeerID:         n.Keys.PeerID,
+		AdminAddr:      admin,
+		MultiAddr:      n.Addr.String(),
+		IsBootstrap:    n.IsBoostrap,
+	}, nil
+}
+
+func (n Node) JDChainConfigs() ([]*nodev1.ChainConfig, error) {
+	var chainConfigs []*nodev1.ChainConfig
+	for _, selector := range n.Chains {
+		family, err := chainsel.GetSelectorFamily(selector)
+		if err != nil {
+			return nil, err
+		}
+
+		// NOTE: this supports non-EVM too
+		chainID, err := chainsel.GetChainIDFromSelector(selector)
+		if err != nil {
+			return nil, err
+		}
+
+		var ocrtype chaintype.ChainType
+		switch family {
+		case chainsel.FamilyEVM:
+			ocrtype = chaintype.EVM
+		case chainsel.FamilySolana:
+			ocrtype = chaintype.Solana
+		case chainsel.FamilyStarknet:
+			ocrtype = chaintype.StarkNet
+		case chainsel.FamilyCosmos:
+			ocrtype = chaintype.Cosmos
+		case chainsel.FamilyAptos:
+			ocrtype = chaintype.Aptos
+		default:
+			return nil, fmt.Errorf("Unsupported chain family %v", family)
+		}
+
+		bundle := n.Keys.OCRKeyBundles[ocrtype]
+		offpk := bundle.OffchainPublicKey()
+		cpk := bundle.ConfigEncryptionPublicKey()
+
+		keyBundle := &nodev1.OCR2Config_OCRKeyBundle{
+			BundleId:              bundle.ID(),
+			ConfigPublicKey:       common.Bytes2Hex(cpk[:]),
+			OffchainPublicKey:     common.Bytes2Hex(offpk[:]),
+			OnchainSigningAddress: bundle.OnChainPublicKey(),
+		}
+
+		var ctype nodev1.ChainType
+		switch family {
+		case chainsel.FamilyEVM:
+			ctype = nodev1.ChainType_CHAIN_TYPE_EVM
+		case chainsel.FamilySolana:
+			ctype = nodev1.ChainType_CHAIN_TYPE_SOLANA
+		case chainsel.FamilyStarknet:
+			ctype = nodev1.ChainType_CHAIN_TYPE_STARKNET
+		case chainsel.FamilyAptos:
+			ctype = nodev1.ChainType_CHAIN_TYPE_APTOS
+		default:
+			panic(fmt.Sprintf("Unsupported chain family %v", family))
+		}
+
+		transmitter := n.Keys.Transmitters[selector]
+
+		chainConfigs = append(chainConfigs, &nodev1.ChainConfig{
+			Chain: &nodev1.Chain{
+				Id:   chainID,
+				Type: ctype,
+			},
+			AccountAddress: transmitter,
+			AdminAddress:   transmitter,
+			Ocr1Config:     nil,
+			Ocr2Config: &nodev1.OCR2Config{
+				Enabled:     true,
+				IsBootstrap: n.IsBoostrap,
+				P2PKeyBundle: &nodev1.OCR2Config_P2PKeyBundle{
+					PeerId: n.Keys.PeerID.String(),
+				},
+				OcrKeyBundle:     keyBundle,
+				Multiaddr:        n.Addr.String(),
+				Plugins:          nil,
```

Comment:
okay to have nil plugins here? For CCIP I know in real node we have commit execute plugin enabled